block-explorer/backend/src/tests/mod.rs

```diff
@@ -0,0 +1,365 @@
+        let apply_block_result = demo.apply_slot(
+            &current_root,
+            Default::default(),
+            &filtered_block.header,
+            &filtered_block.validity_cond,
+            &mut blobs[height as usize],
+        );
+        current_root = apply_block_result.state_root;
+        for receipts in apply_block_result.batch_receipts {
+            data_to_commit.add_batch(receipts);
+        }
+
+        ledger_db.commit_slot(data_to_commit).unwrap();
+        height += 1;
+    }
+
+    ledger_db
+}
+
+async fn create_test_server(
+    docker_cli: &Cli,
+) -> (
+    PostgresContainer,
+    jsonrpsee::server::ServerHandle,
+    TestServer,
+) {
+    use sov_ledger_rpc::server::rpc_module;
+
+    type B = SequencerOutcome<MockAddress>;
+    type Tx = TxEffect;
+
+    let postgres_container = docker_cli.run(PostgresImage::default());
+    let ledger_db = populate_ledger_db();
+    let server = jsonrpsee::server::ServerBuilder::default()
+        .build("127.0.0.1:0")
+        .await
+        .unwrap();
+    let addr = server.local_addr().unwrap();
+    let server_rpc_module = rpc_module::<LedgerDB, B, Tx>(ledger_db).unwrap();
+    let server_handle = server.start(server_rpc_module);
+    let rpc = Arc::new(
+        WsClientBuilder::new()
+            .build(format!("ws://{}", addr))
+            .await
+            .unwrap(),
+    );
+    let connection_string = &format!(
+        "postgresql://postgres:postgres@127.0.0.1:{}/postgres?sslmode=disable",
+        postgres_container.get_host_port_ipv4(5432)
+    );
+    let app_state = Arc::new(AppStateInner {
+        db: Db::new(connection_string).await.unwrap(),
+        rpc,
+        base_url: "http://localhost:3010".to_string(),
+    });
+    index_blocks(app_state.clone(), Duration::default())
+        .await
+        .unwrap();
+    let service = crate::api_v0::router(app_state).into_make_service();
+    (
+        postgres_container,
+        server_handle,
+        TestServer::new(service).unwrap(),
+    )
+}
+
+fn is_sorted<T>(iter: &[T], desired_order: Ordering) -> bool
+where
+    T: Ord + Display,
+{
+    for pair in iter.windows(2) {
+        let a = &pair[0];
+        let b = &pair[1];
+        if a.cmp(b) != desired_order {
+            println!("{} and {} are not in the right order", a, b);
+            return false;
+        }
+    }
+
+    true
+}
+
+// FIX THESE TESTS
+//async fn get_transactions_req(server: &TestServer, query: TransactionsQuery) -> TestResponse {}
+//
+//#[tokio::test]
+//async fn get_transactions_query_combinations(pool: Pool<Postgres>) {
+//    let (_, server) = create_test_server(pool).await;
+//
+//    get_transactions_req(&server, &[("page[selection]", "first")])
+//        .await
+//        .assert_status_ok();
+//    get_transactions_req(&server, &[("page[selection]", "last")])
+//        .await
+//        .assert_status_ok();
+//}
+
+#[tokio::test]
+#[ignore]
+async fn events_first_page_is_full() {
```

Comment:
Is it possible to move the tests to integration tests in `backend/tests`?

Reply:
Done!